services/galley/src/Galley/API/Query.hs

```diff
@@ -313,12 +318,24 @@ getConversationRoles lusr cnv = do
   pure $ Public.ConversationRolesList wireConvRoles
 
 conversationIdsPageFromUnqualified ::
-  Member (ListItems LegacyPaging ConvId) r =>
+  Members
+    [ ListItems LegacyPaging ConvId,
+      ConversationStore,
+      MemberStore,
+      TeamStore
+    ]
+    r =>
   Local UserId ->
   Maybe ConvId ->
   Maybe (Range 1 1000 Int32) ->
   Sem r (Public.ConversationList ConvId)
 conversationIdsPageFromUnqualified lusr start msize = do
+  void $
+    E.getUserTeams (tUnqualified lusr) >>= \tids ->
+      runError @InternalError $
+        runError @(Tagged 'NotATeamMember ())
+          (for_ tids $ \tid -> getGlobalTeamConversation lusr tid)
```

Comment:
Why leave a side effect in listing conversations; which, as a sideeffect of getting global team conversations, creates one?

Why not create the global team conversation during the creation of a *team*?

And if you'd like to enable this functionality for existing teams, you could add a one-off data migration as done in https://github.com/wireapp/wire-server/tree/develop/services/galley/migrate-data/src - read the list of teams, insert one record into conversations, and from then on all teams will have this.

Reply:
I think it would also be possible to use the approach you suggest, but after some discussion with other backend people, I'm not sure it's really worth it.

Ultimately, the "side effect" of creating the global team conversation is not really a side effect, since it's not observable from the outside. The small extra lookup needed when listing conversations doesn't seem like a huge problem, considering that listing conversations is already a slow operation.

On the other hand, preemptively creating all GTCs (and for consistency, all MLS self conversations) forces the creation of all these extra conversations for inactive teams and users. I admit that this is probably not a big concern.

Therefore, we decided to stick with the current approach for now. I'm happy to discuss this further if you still think it's a mistake.